plugins/woocommerce-admin/client/products/components/product-link-field.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Spinner } from '@wordpress/components';
+import { cleanForSlug } from '@wordpress/url';
+import {
+	Product,
+	PRODUCTS_STORE_NAME,
+	WCDataSelector,
+} from '@woocommerce/data';
+import { useFormContext } from '@woocommerce/components';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { EditProductLinkModal } from '../shared/edit-product-link-modal';
+import { getProductPermalinkParts } from '../utils/get-product-permalink-parts';
+import { useProductAutoSave } from '../hooks/use-product-auto-save';
+
+export const ProductLinkField = () => {
+	const { errors, touched, values } = useFormContext< Product >();
+	const [ showProductLinkEditModal, setShowProductLinkEditModal ] =
+		useState( false );
+	const hasNameError = Boolean( touched.name ) && Boolean( errors.name );
+	const { isAutoSaving } = useProductAutoSave( [ 'name' ] );
```

Comment:
This may need to be moved up to a parent component at some point if we have multiple areas that auto-save, but the dependencies should easily allow us to define when we want auto-save to occur.